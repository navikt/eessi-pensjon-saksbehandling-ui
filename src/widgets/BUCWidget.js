import React, { useState, useEffect } from 'react'
import PT from 'prop-types'
import ReactResizeDetector from 'react-resize-detector'
import Buc from 'applications/BUC/widgets/'

const BUCWidget = (props) => {
  const [mounted, setMounted] = useState(false)
  const { onResize } = props

  useEffect(() => {
    if (!mounted && onResize) {
      onResize()
      setMounted(true)
    }
  }, [mounted, onResize])

  return (
    <div className='w-BucWidget'>
      <ReactResizeDetector
        handleWidth
        handleHeight
        onResize={onResize}
      />
      <Buc />
    </div>
  )
}

BUCWidget.properties = {
  type: 'buc',
  title: 'BUC widget',
  description: 'Widget for BUC',
  layout: {
    lg: { minW: 6, maxW: 12, defaultW: 6, minH: 2, defaultH: 4, maxH: 999 },
    md: { minW: 3, maxW: 3, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 },
    sm: { minW: 1, maxW: 1, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 }
  },
  options: {}
}

BUCWidget.propTypes = {
  onResize: PT.func.isRequired
}

export default BUCWidget
