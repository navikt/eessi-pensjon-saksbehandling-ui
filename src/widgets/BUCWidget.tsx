import Buc from 'applications/BUC/'
import { WidgetPropType } from 'declarations/Dashboard.pt'
import { WidgetFC, WidgetProps } from 'eessi-pensjon-ui/dist/declarations/Dashboard'
import _ from 'lodash'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import ReactResizeDetector from 'react-resize-detector'

const BUCWidget: WidgetFC<WidgetProps> = ({
  onResize, onFullFocus, onRestoreFocus, widget
}: WidgetProps): JSX.Element => {
  const [mounted, setMounted] = useState<boolean>(false)

  useEffect(() => {
    if (!mounted && onResize) {
      onResize()
      setMounted(true)
    }
  }, [mounted, onResize])

  return (
    <div className='w-BucWidget'>
      <ReactResizeDetector
        handleWidth
        handleHeight
        onResize={onResize}
      />
      <Buc
        allowFullScreen={_.get(widget, 'options.allowFullScreen')}
        onRestoreFocus={onRestoreFocus}
        onFullFocus={onFullFocus}
      />
    </div>
  )
}

BUCWidget.properties = {
  type: 'buc',
  title: 'BUC widget',
  description: 'Widget for BUC',
  layout: {
    lg: { minW: 6, maxW: 12, defaultW: 6, minH: 2, defaultH: 4, maxH: 999 },
    md: { minW: 3, maxW: 3, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 },
    sm: { minW: 1, maxW: 1, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 }
  },
  options: {
    allowFullScreen: true
  }
}

BUCWidget.propTypes = {
  onResize: PT.func.isRequired,
  onFullFocus: PT.func.isRequired,
  onRestoreFocus: PT.func.isRequired,
  widget: WidgetPropType.isRequired
}

export default BUCWidget
