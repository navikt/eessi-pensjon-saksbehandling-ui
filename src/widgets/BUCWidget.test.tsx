import { WidgetProps } from 'nav-dashboard'
import React from 'react'
import BUCWidget from './BUCWidget'
import { mount, ReactWrapper } from 'enzyme'

jest.mock('applications/BUC/', () => () => (<div className='mock-a-buc' />))

describe('widgets/BUCWidget', () => {
  let wrapper: ReactWrapper
  const initialMockProps: WidgetProps = {
    onFullFocus: jest.fn(),
    onResize: jest.fn(),
    onRestoreFocus: jest.fn(),
    onUpdate: jest.fn(),
    widget: {
      i: 'i',
      type: 'buc',
      title: 'Buc',
      visible: true,
      options: {}
    }
  }

  beforeEach(() => {
    wrapper = mount(<BUCWidget {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('UseEffect: it tries to resize', () => {
    expect(initialMockProps.onResize).toHaveBeenCalled()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.w-BucWidget')).toBeTruthy()
    expect(wrapper.find('.mock-a-buc')).toBeTruthy()
  })

  it('Has properties', () => {
    expect(BUCWidget.properties).toHaveProperty('type')
    expect(BUCWidget.properties).toHaveProperty('title')
    expect(BUCWidget.properties).toHaveProperty('description')
    expect(BUCWidget.properties).toHaveProperty('layout')
    expect(BUCWidget.properties).toHaveProperty('options')
  })
})
