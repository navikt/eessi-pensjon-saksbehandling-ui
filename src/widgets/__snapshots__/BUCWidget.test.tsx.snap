// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`widgets/BUCWidget Renders 1`] = `
<BUCWidget
  highContrast={false}
  labels={Object {}}
  myWidgets={Object {}}
  onDelete={[MockFunction]}
  onFullFocus={[MockFunction]}
  onResize={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onRestoreFocus={[MockFunction]}
  onUpdate={[MockFunction]}
  setMode={[MockFunction]}
  widget={
    Object {
      "i": "i",
      "options": Object {},
      "title": "Buc",
      "type": "buc",
      "visible": true,
    }
  }
>
  <div
    className="w-BucWidget"
  >
    <ResizeDetector
      handleHeight={true}
      handleWidth={true}
      onResize={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <div />
    </ResizeDetector>
    <Component
      onFullFocus={[MockFunction]}
      onRestoreFocus={[MockFunction]}
    >
      <div
        className="mock-a-buc"
      />
    </Component>
  </div>
</BUCWidget>
`;
